src/libstore/binary-cache-store.cc

```diff
@@ -254,6 +256,24 @@ void BinaryCacheStore::addToStore(const ValidPathInfo & info, const ref<std::str
     if (repair || !fileExists(narInfo->url)) {
         stats.narWrite++;
         upsertFile(narInfo->url, *narCompressed);
+#if ENABLE_IPFS
+        if (publishToIPFS) {
+          try {
+            auto narPath = narInfo->url;
+            auto pos = narInfo->url.find_last_of("/");
+            if (pos != std::string::npos) {
+              narPath = narInfo->url.substr(pos+1);
+            }
+            ipfsHash = IPFSAccessor::addFile(narPath, *narCompressed);
+
+            if (!ipfsHash.empty()) {
```

Comment:
In any case, it shouldn't -- an `IpfsNode` should emit the [hash + name](https://github.com/ipfs/go-ipfs/blob/416f02508475ed4afce726c71f98aad83ad0d73f/core/coreunix/add.go#L480-L481) to the client when the process completes *without an err*.